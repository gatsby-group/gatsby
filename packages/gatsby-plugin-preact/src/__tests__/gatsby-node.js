const { onCreateWebpackConfig } = require(`../gatsby-node`)

describe(`gatsby-plugin-preact`, () => {
  it(`sets the correct webpack config`, () => {
    const actions = { setWebpackConfig: jest.fn() }
    onCreateWebpackConfig({ stage: ``, actions })
    onCreateWebpackConfig({ stage: `develop-html`, actions })
    expect(actions.setWebpackConfig).toHaveBeenCalledTimes(1)
    expect(actions.setWebpackConfig.mock.calls).toMatchSnapshot()
  })
})
