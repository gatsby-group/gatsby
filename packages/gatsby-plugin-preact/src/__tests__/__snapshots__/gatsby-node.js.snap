// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-plugin-preact sets the correct webpack config 1`] = `
Array [
  Array [
    Object {
      "resolve": Object {
        "alias": Object {
          "create-react-class": "preact-compat/lib/create-react-class",
          "react": "preact-compat",
          "react-dom": "preact-compat",
        },
      },
    },
  ],
]
`;
